db/commitlog/commitlog_entry.hh

```diff
@@ -37,9 +37,35 @@ public:
     frozen_mutation&& mutation() && { return std::move(_mutation); }
 };
 
+namespace detail {
+class force_sync : public std::optional<db::timeout_clock::time_point> {
+    using opt_type = std::optional<db::timeout_clock::time_point>; 
+public:
+    constexpr force_sync(opt_type t = std::nullopt)
+        : opt_type(t)
+    {}
+    constexpr force_sync(db::timeout_clock::time_point t)
+        : opt_type(t)
+    {}
+    constexpr force_sync(bool b)
+        : opt_type(b ? yes : no)
+    {}
+
+    bool operator==(const force_sync& v) const {
+        return std::equal_to<opt_type>{}(*this, v);
+    }
+    bool operator!=(const force_sync& v) const {
+        return !(*this == v);
+    }
+
+    static const force_sync no;
+    static const force_sync yes;
+};
+}
+
```

Comment:
Maybe this can be simplified: 

  time_point::min() -> must sync immediately
  time::point(1234) -> must sync by time 1234
  time_point::max() -> totally relaxed (but of course periodic mode will sync it earlier)

It makes sense to do this if it works out naturally, not if we have to add a bunch of ifs everywhere.

Reply:
I see that it's already halfway like this - force_sync::yes is time_point{0}. And maybe max() isn't the same as force_sync::no, because force_sync::no doesn't wait for the sync.

Reply:
Indeed.